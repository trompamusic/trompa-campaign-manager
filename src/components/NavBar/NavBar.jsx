import React, { useState } from 'react';
import * as PropTypes from 'prop-types';
import { Link, NavLink } from 'react-router-dom';
import { makeStyles } from '@material-ui/styles';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import CloseIcon from '@material-ui/icons/Close';
import SwipeableDrawer from '@material-ui/core/SwipeableDrawer';
import AppbarTop from '../AppbarTop/AppbarTop';
import images from '../../theme/images';
import styles from './NavBar.styles';

const useStyles = makeStyles(styles);

export default function NavBar ({
  navLinks,
  buttons,
  primaryButton,
}) {
  const [drawerOpen, setDrawerOpen] = useState(false);
  const isMobile                    = useMediaQuery(theme => theme.breakpoints.down('sm'));

  const classes = useStyles();

  const renderLinks = () => {
    return navLinks?.map(({ name, to }) => (
      <NavLink
        key={name}
        className={classes.navLink}
        activeClassName={classes.navLinkActive}
        to={to}
      >
        {name}
      </NavLink>
    ));
  };

  const renderButtons = () => {
    return buttons?.map(({ name, onClick, startIcon }) => (
      <Button
        key={name}
        classes={{ root: classes.removeHover, text: classes.navLink }}
        component="button"
        onClick={onClick}
        variant="text"
        startIcon={startIcon}
        disableFocusRipple
        disableRipple
      >
        {name}
      </Button>
    ));
  };

  const renderPrimaryButton = () => {
    return primaryButton
       && (
         <Button
           key={primaryButton?.name}
           component={primaryButton?.to ? Link : 'button'}
           to={primaryButton?.to ? primaryButton?.to : undefined }
           onClick={primaryButton?.onClick ? primaryButton?.onClick : undefined }
           variant="contained"
           color="primary"
         >
           {primaryButton?.name}
         </Button>
       );
  };

  return (
    <React.Fragment>
      <div className={classes.mobile}>
        <AppbarTop position="fixed" mobile>
          <IconButton onClick={() => setDrawerOpen(true)} aria-label="menu" component="span">
            <MenuIcon />
          </IconButton>
        </AppbarTop>
        <SwipeableDrawer
          open={drawerOpen}
          onClose={() => setDrawerOpen(false)}
          onOpen={() => setDrawerOpen(true)}
        >
          <div className={classes.drawerTop}>
            <img className={classes.drawerLogo} src={images.logoAsSvg} alt="" />
            <IconButton onClick={() => setDrawerOpen(false)} aria-label="" color="primary" component="span">
              <CloseIcon />
            </IconButton>
          </div>
          <div className={classes.drawerBody}>
            {renderLinks()}
          </div>
          <div className={classes.drawerBottom}>
            {renderButtons()}
            {renderPrimaryButton()}
          </div>
        </SwipeableDrawer>
      </div>
      {!isMobile && (
        <div className={classes.desktop}>
          <AppbarTop position="fixed">
            <div>
              {renderLinks()}
              {renderButtons()}
              {renderPrimaryButton()}
            </div>
          </AppbarTop>
        </div>                )}
    </React.Fragment>
  );
}

NavBar.propTypes = {
  navLinks     : PropTypes.array,
  primaryButton: PropTypes.object,
  drawerContent: PropTypes.element,
};
