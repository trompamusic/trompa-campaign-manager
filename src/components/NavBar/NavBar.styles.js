import { createStyles } from '@material-ui/styles';

export default ({ typography, palette, spacing, breakpoints }) => createStyles({
  navLink: {
    fontSize      : typography.pxToRem(15.8),
    letterSpacing : 0.5,
    textDecoration: 'none',
    color         : palette.text.darkGrey,
    fontFamily    : typography.fontFamily,
    marginRight   : spacing(4),
    padding       : 0,
    '& span'      : {
      paddingBottom: spacing(0.3),
      paddingLeft  : '5px',
    },
    [breakpoints.down('md')]: {
      padding: spacing(0.75),
    },
  },
  removeHover: {
    '&:hover': {
      backgroundColor: 'transparent',
    },
  },
  navLinkActive: {
    color: palette.primary.main,
  },
  mobile: {
    [breakpoints.up('md')]: {
      display: 'none',
    },
  },
  // desktop: {
  //   [breakpoints.down('sm')]: {
  //     display: 'none',
  //   },
  // },
  drawerTop: {
    display        : 'flex',
    justifyContent : 'space-between',
    alignItems     : 'center',
    padding        : spacing(2),
    width          : 288,
    boxSizing      : 'border-box',
    backgroundColor: palette.common.faintGrey,
  },
  drawerBody: {
    display      : 'flex',
    flexDirection: 'column',
    flex         : '1 0 auto',
    padding      : spacing(2, 3),
  },
  drawerBottom: {
    display      : 'flex',
    flexDirection: 'column',
    padding      : spacing(2, 3),
  },
  drawerLogo: {
    width: 138,
  },
});
