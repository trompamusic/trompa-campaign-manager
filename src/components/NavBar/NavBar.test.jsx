import React from 'react';
import { createMemoryHistory } from "history";
import { Router } from "react-router";
import { fireEvent } from '@testing-library/react';
import { render } from '../../testUtils';
import NavBar from './NavBar';

describe('<NavBar />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render((
      <NavBar
        navLinks={[
          { name: 'Home', to: '/' },
        ]}
        primaryButton={{
          name: 'Join campaign', to: 'campaign/e63fc3c5-f84e-4a64-9d5b-98a49dd4680c',
        }}
      />
    ));

    expect(container.firstChild).toMatchSnapshot();
  });

  test('displays navLinks and goes to correct route', () => {
    const history       = createMemoryHistory();
    const { getByText } = render((
      <Router history={history}>
        <NavBar
          navLinks={[
            { name: 'Home', to: 'test' },
          ]}

        />
      </Router>
    ));

    fireEvent.click(getByText('Home'));
    expect(getByText('Home')).toBeTruthy();
    expect(history.location.pathname).toBe("/test");
  });

  test('displays primary button and goes to correct route', () => {
    const history       = createMemoryHistory();
    const { getByText } = render((
      <Router history={history}>
        <NavBar
          primaryButton={{
            name: 'Join campaign', to: 'test',
          }}

        />
      </Router>
    ));

    fireEvent.click(getByText('Join campaign'));
    expect(getByText('Join campaign')).toBeTruthy();
    expect(history.location.pathname).toBe("/test");
  });
});
