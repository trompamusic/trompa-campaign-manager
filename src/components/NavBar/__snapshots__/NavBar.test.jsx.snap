// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders and matches snapshot 1`] = `
<div
  class="makeStyles-mobile-4"
>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-dense makeStyles-dense-9 makeStyles-mobile-17 MuiToolbar-gutters"
    >
      <div
        class="makeStyles-hamburger-18"
      >
        <span
          aria-disabled="false"
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
      </div>
      <a
        href="/"
        rel="noopener noreferrer"
      >
        <img
          alt="Trompa logo"
          class="makeStyles-logo-11"
          src="[object Object]"
        />
      </a>
    </div>
  </header>
  <div
    class="PrivateSwipeArea-root-134 PrivateSwipeArea-anchorLeft-135"
    style="width: 20px;"
  />
</div>
`;
